File: js/programmers/level.0/prgm_120861_1.js
function solution(keyinput, board) {
  let result = [0, 0];
  const row = board[0] / 2;
  const column = board[1] / 2;

  for (i = 0; i < keyinput.length; i++) {
    if (keyinput[i] === "right" && result[0] + 1 < row) result[0]++;
    if (keyinput[i] === "left" && result[0] - 1 > -row) result[0]--;
    if (keyinput[i] === "up" && result[1] + 1 < column) result[1]++;
    if (keyinput[i] === "down" && result[1] - 1 > -column) result[1]--;
  }
  return result;
}

File: python/programmers/level.0/prgm_120861_1.py
def solution(keyinput, board):
    result = [0, 0]
    row = board[0] / 2
    column = board[1] / 2

    for i in range(len(keyinput)):
        if keyinput[i] == "left" and result[0] - 1 > -row:
            result[0] -= 1
        if keyinput[i] == "right" and result[0] + 1 < row:
            result[0] += 1
        if keyinput[i] == "up" and result[1] + 1 < column:
            result[1] += 1
        if keyinput[i] == "down" and result[1] - 1 > -column:
            result[1] -= 1
            
    return result